### PR TITLE
feat: Add workaround for CSRF SameSite=None cookies

### LINKED ISSUE #1753: Workaround needed for CSRF SameSite=None cookies
It's really good that Hydra is already supporting cookies with `SameSite=None` from https://github.com/ory/hydra/pull/1718. However, it seems like some workarounds are needed for some not-so-old browsers.

The entire issue is described here: https://auth0.com/blog/browser-behavior-changes-what-developers-need-to-know/#Working.around.incompatible.browsers

From my understanding of the workaround, when `SameSite=None` is set, a second cookie must also be set without the SameSite attribute, and both must be checked when doing the CSRF validation, as only one of them is likely to actually be available.

I think I'll have time this week to submit a PR if this sounds good.

### PR DESCRIPTION
**Pull request by jcxplorer**
## Related issue

Closes #1753 as discussed with @aeneasr.

## Proposed changes

Implements the workaround from https://web.dev/samesite-cookie-recipes/ for the CSRF cookies only when using SameSite=None. This is configurable and disabled by default.

Also adds some unit tests for the existing CSRF cookie helpers, along with unit tests for this change.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have read the [security policy](../security/policy).
- [xI confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](docs/docs).

## Further comments

Regenerating the documentation changed quite a few bits in the configuration section that were unrelated to my change. Please double check those changes 😬

### CODE REVIEW

**Comment by jcxplorer** [comment]
The nancy test runs fine locally, so probably just needs to be re-triggered in the CI. Seems like the webservice it uses was down at the time.

**Comment by aeneasr** [comment]
Yup, triggering CI

File: consent/helper_test.go [hunk 1]
```diff
@@ -95,3 +99,264 @@ func TestMatchScopes(t *testing.T) {
...
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: false,
+			expectError:              false,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
```
Comment (aeneasr) [inline consent/helper_test.go:221]: Shouldn't this be `http.SameSiteNoneMode`?

File: consent/helper_test.go [hunk 2]
```diff
@@ -95,3 +99,264 @@ func TestMatchScopes(t *testing.T) {
...
+			csrfValue:                "CSRF-VALUE",
+			sameSiteLegacyWorkaround: true,
+			expectError:              false,
+		},
+		{
+			cookies: []cookie{
+				{
+					name:      cookieAuthenticationCSRFName,
+					csrfValue: "CSRF-VALUE",
+					sameSite:  http.SameSiteDefaultMode,
```
Comment (aeneasr) [inline consent/helper_test.go:204]: Shouldn't this be `http.SameSiteNoneMode`?

File: consent/helper_test.go (hunk 1 above)
Comment (jcxplorer) [reply consent/helper_test.go:221]: Yup, will update.

**Comment by jcxplorer** [comment]
Great, I'll make the few changes and rebase against master. @aeneasr do you want me to amend the current commit or have a separate commit?

**Comment by aeneasr** [comment]
I'm squash-merging PRs always anyways so two commits is fine!

**Comment by jcxplorer** [comment]
All done! 🙌

**Comment by aeneasr** [comment]
Thank you!

**Pull request merged**